dbms/src/Functions/FunctionsDateTime.h

```diff
@@ -3313,6 +3313,93 @@ struct TiDBToSecondsTransformerImpl
     }
 };
 
+template <typename ToFieldType>
+struct TiDBToDaysTransformerImpl
+{
+    static constexpr auto name = "tidbToDays";
+
+    static void execute(const Context & context,
+                        const ColumnVector<DataTypeMyTimeBase::FieldType>::Container & vec_from,
+                        typename ColumnVector<ToFieldType>::Container & vec_to,
+                        typename ColumnVector<UInt8>::Container & vec_null_map)
+    {
+        bool is_null = false;
+        for (size_t i = 0; i < vec_from.size(); ++i)
+        {
+            MyTimeBase val(vec_from[i]);
+            vec_to[i] = execute(context, val, is_null);
+            vec_null_map[i] = is_null;
+            is_null = false;
+        }
+    }
+
+    static ToFieldType execute(const Context & context, const MyTimeBase & val, bool & is_null)
+    {
+        // TiDB returns normal value if one of month/day is zero for to_seconds function, while MySQL return null if either of them is zero.
+        // TiFlash aligns with MySQL to align the behavior with other functions like last_day.
+        if (val.month == 0 || val.day == 0)
+        {
+            context.getDAGContext()->handleInvalidTime(
+                    fmt::format("Invalid time value: month({}) or day({}) is zero", val.month, val.day),
+                    Errors::Types::WrongValue);
+            is_null = true;
+            return 0;
+        }
+        return static_cast<ToFieldType>(calcDayNum(val.year, val.month, val.day));
+    }
+};
+
+class FunctionTiDBFromDays : public IFunction
+{
+public:
+    static constexpr auto name = "tidbConcat";
+
+    explicit FunctionTiDBFromDays(const Context &){}
+
+    static FunctionPtr create(const Context & context) { return std::make_shared<FunctionTiDBFromDays>(context); }
+
+    String getName() const override { return name; }
+    bool isVariadic() const override { return false; }
+    size_t getNumberOfArguments() const override { return 0; }
+    bool useDefaultImplementationForNulls() const override { return true; }
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (arguments.size() != 1)
```

Comment:
Don't need to check the argument number here if `isVariadic` is false.

Reply:
updated